keps/sig-node/3695-pod-resources-for-dra/README.md

```diff
@@ -0,0 +1,434 @@
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `DynamicResourceAllocation` is existing feature gate to
+  enable / disable DRA feature.
+  - Components depending on the feature gate: kube-apiserver, kube-controller-manager,
+  kube-scheduler, kubelet
+  - Feature gate name: `PodResourcesDynamicResources` new feature gate to
+  enable / disable PodResources API List method to populate `DynamicResource`
+  information from the `DRAManager`.
+  `DynamicResourceAllocation` feature gate has to be enabled as well.
+  - Components depending on the feature gate: kubelet, 3rd party consumers.
+  - Feature gate name: `PodResourcesGet` new feature gate to enable / disable
+    PodResources API Get method. In case `DynamicResourceAllocation` or
+    the `PodResourcesDynamicResources` are disabled and `PodResourcesGet`
+    is enabled, the Get method will retrieve resources allocated by device plugins,
+    memory and cpus (but omit those allocated by DRA resource drivers).
+    In case `PodResourcesGet`, `DynamicResourceAllocation` and `PodResourcesDynamicResources`
+    are all enabled, the `Get()` method will also retrieve the resources allocated via DRA.
+  - Components depending on the feature gate: kubelet, 3rd party consumers.
+
+###### Does enabling the feature change any default behavior?
+
+No.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, through feature gates.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The API becomes available again. The API is stateless, so no recovery is needed, clients can just consume the data.
+
+###### Are there any tests for feature enablement/disablement?
+
+e2e test will demonstrate that when the feature gate is disabled, the API returns the appropriate error code.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Kubelet may fail to start. The new API may report inconsistent data, or may cause the kubelet to crash.
+
+###### What specific metrics should inform a rollback?
+
+`pod_resources_endpoint_errors_get` - but only with feature gate `PodResourcesGet` enabled. Otherwise the API will always return a known error.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Not Applicable.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+Look at the `pod_resources_endpoint_requests_list` and `pod_resources_endpoint_requests_get` metric exposed by the kubelet.
+
+###### How can someone using this feature know that it is working for their instance?
```

Comment:
"this instance" is referring to a specific use of a resource. in this case, the feature isn't really controlled by a user defined resource - it's either on or off for all instances, so you probably don't really need to answer this question. But a reasonable answer would be to call the API for the pod in question and see the result - that is aligned with the spirit of the question and what a user might do to check if it's working for them while troubleshooting.